Assets/ModScripts/Submodules/LyingWires.cs

```diff
@@ -0,0 +1,390 @@
+        colorConditions.Add(3, Info.Button == 8 || Info.Button == 3 || Info.Button == 1);
+        colorConditions.Add(4, Info.Button == 9 || Info.Button == 0);
+        colorConditions.Add(5, Info.Button == 1 || Info.Button == 7 || Info.Button == 6 || Info.Button == 0 || Info.Button == 9);
+        colorConditions.Add(6, Info.Button == 8 || Info.Button == 5 || Info.Button == 10);
+        colorConditions.Add(7, true);
+        colorConditions.Add(8, Info.ButtonText == "YES" || Info.ButtonText == "NO" || Info.ButtonText == "I DON'T KNOW");
+        colorConditions.Add(9, Info.ButtonText.Length < 5);
+        colorConditions.Add(10, Info.MainColors[Info.Button].ToUpper().Contains("R"));
+        colorConditions.Add(11, Info.ButtonText == "PRESS" || Info.ButtonText == "TAP" || Info.ButtonText == "PUSH" || Info.ButtonText == "CLICK");
+
+        colorStatements.Add(0, "The button’s label begins with a letter N-Z");
+        colorStatements.Add(1, "The button is Blue");
+        colorStatements.Add(2, "The name of the button’s color ends with a Letter A-M");
+        colorStatements.Add(3, "The button is Red, Green, or Blue");
+        colorStatements.Add(4, "The button is White or Black");
+        colorStatements.Add(5, "The button’s color can be found in the right half of the manual's first table");
+        colorStatements.Add(6, "The button is Red, Orange, or Yellow");
+        colorStatements.Add(7, "The button is a button");
+        colorStatements.Add(8, "The button’s label is “YES”, “NO”, or “I DON'T KNOW”");
+        colorStatements.Add(9, "The button’s label contains fewer than 5 letters");
+        colorStatements.Add(10, "The button’s color name contains the letter “R”");
+        colorStatements.Add(11, "The button’s label is “Press”, “Tap”, “Push”, or “Click”");
+
+        trueColors = colorConditions.Where(x => x.Value).Select(x => x.Key).ToArray();
+        UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] The button is {1}.", ModuleID, Info.GetButtonInfo());
+        UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] The wires are as follows: {1}.", ModuleID, Info.GetWireInfo());
+        UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] The wire LEDs are as follows: {1}.", ModuleID, Info.GetWireLEDInfo());
+        DetermineWires();
+    }
+
+    public override void OnWireCut(int Wire)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.WireSnip, Module.transform);
+        Module.CutWire(Wire);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Strike! Wire {1} was cut when the component selection was [{2}] instead of [{3}].", ModuleID, Wire + 1, Module.GetOnComponents(), Module.GetTargetComponents());
+            RegenWires();
+            return;
+        }
```

Comment:
```suggestion
        if (!Module.IsSolving())
{
    if (!Module.CheckValidComponents())
    {
        UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Strike! Wire {1} was cut when the component selection was [{2}] instead of [{3}].", ModuleID, Wire + 1, Module.GetOnComponents(), Module.GetTargetComponents());
        RegenWires();
        return;
    }

    Module.StartSolve();
}
```
Optimizes the component check and the StartSolve call, which needs to be called immediately once the component check passes. Be sure to remove the existing StartSolve call at the end of this method